srtcore/buffer.cpp

```diff
@@ -63,6 +63,41 @@ using namespace std;
 using namespace srt_logging;
 using namespace srt::sync;
 
+bool AvgBufSize::canUpdate(const time_point& now) const
+{
+    const int      SRT_MAVG_BASE_PERIOD = 1000000; // us
+    const int      SRT_us2ms            = 1000;
+    const uint64_t elapsed_ms           = count_milliseconds(now - m_tsLastSamplingTime); // ms since last sampling
+    return (elapsed_ms >= (SRT_MAVG_BASE_PERIOD / SRT_MAVG_SAMPLING_RATE) / SRT_us2ms);
+}
+
+void AvgBufSize::update(const steady_clock::time_point& now, int pkts, int bytes, int timespan_ms)
+{
+    const uint64_t elapsed_ms = count_milliseconds(now - m_tsLastSamplingTime); // ms since last sampling
+
+    m_tsLastSamplingTime = now;
+
+    if (elapsed_ms > 1000)
+    {
+        // No sampling in last 1 sec, initialize average
+        m_dCountMAvg      = pkts;
+        m_dBytesCountMAvg = bytes;
+        m_dTimespanMAvg   = timespan_ms;
+        return;
+    }
+
+    //
+    // weight last average value between -1 sec and last sampling time (LST)
+    // and new value between last sampling time and now
+    //                                      |elapsed_ms|
+    //   +----------------------------------+-------+
+    //  -1                                 LST      0(now)
+    //
+    m_dCountMAvg      = avg_iir_w<1000>(m_dCountMAvg, static_cast<double>(pkts), elapsed_ms);
```

Comment:
Doing `avg_iir_w<1000, double>(m_dCountMAvg, pkts, elapsed_ms)` should give you the same result.

Reply:
Right, good point.